openqa-maintenance.py

```diff
@@ -88,6 +90,42 @@ def settings(self, src_prj, dst_prj, packages, req=None):
         return self._settings.copy()
 
 
+class SUSEUpdate(Update):
+    def __init__(self, settings):
+        Update.__init__(self, settings)
+
+    # take the first package name we find - often enough correct
+    def request_name(self, req):
+        if request_name_cache.get(req.reqid):
+            return request_name_cache[req.reqid]
+
+        request_name_cache[req.reqid] = 'unknown'
+        for action in  req.get_actions('maintenance_release'):
+            if action.tgt_package.startswith('patchinfo'):
+                continue
+            print action.src_package
+            url = osc.core.makeurl(
+                req.apiurl,
+                ('source', action.src_project, action.src_package, '_link'))
+            root = ET.parse(osc.core.http_GET(url)).getroot()
+            if root.attrib.get('cicount'):
+                continue
+            request_name_cache[req.reqid] = action.tgt_package
+            break
+
+        return request_name_cache[req.reqid]
```

Comment:
```python
if req.reqid not in request_name_cache:
    request_name_cache[req.reqid] = self._request_name(req)
return request_name_cache[req.reqid]
```

and in `def _request_name(self, req)`

```python
name = 'unknown'
actions = [action for action in req.get_actions('maintenance_release')
           if not action.tgt_package.startswith('patchinfo')]

for action in actions:
    url = osc.core.makeurl(
        req.apiurl,
        ('source', action.src_project, action.src_package, '_link'))
    root = ET.parse(osc.core.http_GET(url)).getroot()
    if root.attrib.get('cicount'):
        continue
    name = action.tgt_package
    break

return name
```

Can be shorted if there is only one instance of such an element